lib/output.js

```diff
@@ -1444,17 +1447,20 @@ function OutputStream(options) {
         }
     });
     DEFPRINT(AST_ObjectKeyVal, function(self, output){
-        var print_as_shorthand = self.shorthand &&
+        var value;
+        if (output.option("shorthand") &&
             self.value instanceof AST_Symbol &&
-            self.key == self.value.print_to_string();
-
-        if (print_as_shorthand) {
-            output.print_name(self.key);
-            return;
+            is_identifier_string(self.key) &&
+            (value = self.value.definition()) &&
```

Comment:
definition() depends on figure_out_scope().

Consider changing `value =` to `def =`.